File: DroidsAPI/src/main/java/com/github/rochedo098/droidsapi/api/script/classes/RobotActions.java
package com.github.rochedo098.droidsapi.api.script.classes;

import com.github.rochedo098.droidsapi.api.script.classes.listeners.RobotActionsListener;
import dev.latvian.mods.rhino.ScriptableObject;
import dev.latvian.mods.rhino.annotations.JSFunction;
import dev.latvian.mods.rhino.annotations.JSGetter;
import org.jetbrains.annotations.ApiStatus;

import java.util.ArrayList;
import java.util.List;

public class RobotActions extends ScriptableObject {
    private final List<RobotActionsListener> listeners = new ArrayList<>();

    public void addListener(RobotActionsListener toAdd) {
        listeners.add(toAdd);
    }

    @Override
    public String getClassName() { return "RobotActions"; }

    @JSFunction
    public void toUse(String blockPos) {
        for (RobotActionsListener ral : listeners) {
            ral.toUse(blockPos);
        }
    }

    @JSFunction
    public void toBreak(String blockPos) {
        for (RobotActionsListener ral : listeners) {
            ral.toBreak(blockPos);
        }
    }

    @ApiStatus.Experimental
    @JSGetter
    public String checkBlock(String blockPos) {
        for (RobotActionsListener ral : listeners) {
            return ral.checkBlock(blockPos);
        }
        return "null";
    }
}

File: src/main/kotlin/com/github/rochedo098/droids/api/script/classes/RobotActions.kt
package com.github.rochedo098.droids.api.script.classes

import com.github.rochedo098.droids.api.script.classes.listeners.RobotActionsListener
import dev.latvian.mods.rhino.ScriptableObject
import dev.latvian.mods.rhino.annotations.JSFunction
import dev.latvian.mods.rhino.annotations.JSGetter
import org.jetbrains.annotations.ApiStatus

class RobotActions : ScriptableObject() {
    private val listeners: MutableList<RobotActionsListener> = ArrayList<RobotActionsListener>()
    fun addListener(toAdd: RobotActionsListener) {
        listeners.add(toAdd)
    }

    override fun getClassName(): String = "RobotActions"

    @JSFunction
    fun toUse(blockPos: String?) {
        for (ral in listeners) {
            ral.toUse(blockPos)
        }
    }

    @JSFunction
    fun toBreak(blockPos: String?) {
        for (ral in listeners) {
            ral.toBreak(blockPos)
        }
    }

    @ApiStatus.Experimental
    @JSGetter
    fun checkBlock(blockPos: String?): String? {
        for (ral in listeners) {
            return ral.checkBlock(blockPos)
        }
        return "null"
    }
}